support-frontend/assets/components/subscriptionCheckouts/billingPeriodSelector.jsx

```diff
@@ -0,0 +1,67 @@
+// @flow
+import React from 'react';
+
+import { Fieldset } from 'components/forms/fieldset';
+import { RadioInputWithHelper } from 'components/forms/customFields/radioInputWithHelper';
+import type { BillingPeriod } from 'helpers/billingPeriods';
+import {
+  billingPeriodTitle,
+  Quarterly,
+  SixWeekly,
+} from 'helpers/billingPeriods';
+import { FormSection } from 'components/checkoutForm/checkoutForm';
+import type { ProductPrices } from 'helpers/productPrice/productPrices';
+import { getProductPrice } from 'helpers/productPrice/productPrices';
+import type { IsoCountry } from 'helpers/internationalisation/country';
+import type { FulfilmentOptions } from 'helpers/productPrice/fulfilmentOptions';
+import { NoFulfilmentOptions } from 'helpers/productPrice/fulfilmentOptions';
+import type { Action } from 'helpers/subscriptionsForms/formActions';
+import { extendedGlyphForCountry } from 'helpers/internationalisation/currency';
+import {
+  getAppliedPromoDescription,
+  getPriceDescription,
+} from 'helpers/productPrice/priceDescriptions';
+
+type PropTypes = {|
+  productPrices: ProductPrices,
+  billingPeriods: BillingPeriod[],
+  fulfilmentOption?: FulfilmentOptions,
+  billingCountry: IsoCountry,
+  selected: BillingPeriod,
+  onChange: (BillingPeriod) => Action,
+|}
+
+function BillingPeriodSelector(props: PropTypes) {
+  return (
+    <FormSection title="How often would you like to pay?">
+      <Fieldset legend="How often would you like to pay?">
+        {props.billingPeriods.map((billingPeriod) => {
+          const productPrice = getProductPrice(
+            props.productPrices,
+            props.billingCountry,
+            billingPeriod === SixWeekly ? Quarterly : billingPeriod, // for 6 for 6 we need the quarterly pricing
```

Comment:
Is this because 6 for 6 is really a quarterly order with a discount applied?

Reply:
Yes that's right, it's all a bit messy unfortunately

Reply:
👍